src/lib/config/ModelConfig.cc

```diff
@@ -424,15 +424,24 @@ void ModelConfig::setExpectations(bool isDefault) {
   expectations_.addChild(
       ExpectationNode::createExpectation("Branch-Predictor"));
 
+  expectations_["Branch-Predictor"].addChild(
+      ExpectationNode::createExpectation<std::string>("Perceptron", "Type"));
+  expectations_["Branch-Predictor"]["Type"].setValueSet(
+      std::vector<std::string>{"Generic", "Perceptron"});
+
   expectations_["Branch-Predictor"].addChild(
       ExpectationNode::createExpectation<uint8_t>(8, "BTB-Tag-Bits"));
   expectations_["Branch-Predictor"]["BTB-Tag-Bits"].setValueBounds<uint8_t>(1,
                                                                             64);
-
-  expectations_["Branch-Predictor"].addChild(
-      ExpectationNode::createExpectation<uint8_t>(2, "Saturating-Count-Bits"));
-  expectations_["Branch-Predictor"]["Saturating-Count-Bits"]
-      .setValueBounds<uint8_t>(1, 64);
+  // Saturating counter bits are relevant to the GenericPredictor only
+  if (!isDefault &&
```

Comment:
We could combine the two Generic-only config options under one conditional to reduce code duplication